tracee-ebpf/main.go

```diff
@@ -1077,6 +1095,23 @@ func getBPFObject() (string, error) {
 	return bpfObjFilePath, nil
 }
 
+func unpackCOREBinary() (string, error) {
+	b, err := bpfBundleInjected.ReadFile("dist/tracee.bpf.core.o")
+	if err != nil {
+		return "", err
+	}
+	f, err := ioutil.TempFile("/tmp/tracee", "tracee.bpf.core.o")
```

Comment:
I'm not sure I understand your feedback. /tmp/tracee is just a place to intermediately place the core object when unloading it from the embedded bytes. One alternative is to rewire tracee to load the object from memory and use libbpfgo's `NewModuleFromBuffer` instead.

Reply:
I'll try to elaborate. Let me know if I misunderstood something:
There are 2 special directories for Tracee: "installPath" and "outputPath". both are user-configurable, and both default to `/tmp/tracee`. Nothing in Tracee writes directly to `/tmp/tracee`, we always write either to "installPath" or "outputPath". In this case, I think you meant to put the CORE object file under "installPath".
Now what I was saying is that this "installPath" is a permanent directory (the fact that it defaults to /tmp is confusing.. it's not very temporary in its nature). So my feedback was that for that reason I don't think it's a good "place to intermediately place the core object when unloading it from the embedded bytes". 
But even if you did want to place the intermediary CORE file in the "installPath", the code is missing a few things:
1. use the variable name (installPath), instead of hardcoding the default path
2. ensure that the path exists (like we do in other cases where we write there), otherwise it will fail
3. clean after it (if it's really an intermediary file it should be gone as soon as possible).
I also think that if it's really an intermediary file, it doesn't matter where it is and it's better to let `TempFile` generate a path under `/tmp`

In the same comment I was also wondering if we really need to "unpack" the embedded file to a temp file at all, or we can just `Open` it directly from `embed.FS`. I think libbpfgo should accept an `io.reader` instead (or in addition) to `[]bytes` as this is better Go code, but if it's expanding the scope of the PR let's leave it for some other time.

Reply:
Ah ok this all makes sense. I'm going to change this to just use the file system's default tempfs. 

Regarding not writing to file at all - you're correct that there's a way to not write it to a file at all. The best way to do this would be to read it all into memory and load it directly with libbpf. When originally writing this I didn't think it was important to rewire tracee to support it but i'm open to it if you feel it's be a significant improvement.